src/app/services/mail.service.ts

```diff
@@ -0,0 +1,124 @@
+import dedent from 'dedent-js'
+import Mail from 'nodemailer/lib/mailer'
+import validator from 'validator'
+import { Logger } from 'winston'
+
+import config from '../../config/config'
+import { createLoggerWithLabel } from '../../config/logger'
+import { HASH_EXPIRE_AFTER_SECONDS } from '../../shared/util/verification'
+import { EMAIL_HEADERS, EMAIL_TYPES } from '../utils/constants'
+
+const mailLogger = createLoggerWithLabel('mail')
+
+type SendMailOptions = {
+  mailId?: string
+  formId?: string
+}
+
+type MailServiceParams = {
+  appName?: string
+  transporter?: Mail
+  senderEmail?: string
+  logger?: Logger
+}
+
+export class MailService {
+  #appName: string
+  #transporter: Mail
```

Comment:
Is there a reason why you're using the `#` syntax instead of `private`? Personally I prefer `private` as it's more explicit

Reply:
`private` in TypeScript is only for TypeScript compiler, but `#` is native Javascript's private modifier. Meaning the Javascript code still can access `MailService.appName` (and be valid), and since we have both JS and TS files intermixed it is safer to use `#`.

https://www.valentinog.com/blog/private/

PS: His argument against `#` doesn't stand since this is serverside code (hopefully)